File: medium/top_frequent_words.py
def top_frequent_words(words: list[str], k: int) -> list[str]:
    freq = {}
    for word in words:
        freq[word] = freq.get(word, 0) + 1
    result = sorted(freq.items(), key=lambda x: (-x[1], x[0]))
    return list(map(lambda x: x[0], result))[0:k]

File: src/main/php/Medium/TopFrequentWords.php
<?php

declare(strict_types=1);

namespace Src\Medium;

class TopFrequentWords
{
    /**
     * @param  string[]  $words
     * @return string[]
     */
    public function solve(array $words, int $k): array
    {
        $result = [];
        foreach ($words as $word) {
            $result[$word] = ($result[$word] ?? 0) + 1;
        }
        uksort($result, 'strcmp');
        uasort($result, fn ($a, $b) => $b - $a);

        return array_slice(array_keys($result), 0, $k);
    }
}
